internal/resp/decoder/errors.go

```
@@ -18,6 +18,11 @@ type InvalidInputError struct {
 	Message string
 }
 
+type InvalidRESPError struct {
```

@UdeshyaDhungana how is `InvalidRESPError` different from `InvalidInputError`? The idea behind the errors here was that there are two ways in which we can fail to parse RESP: either (a) the input is invalid (i.e. a protocol error), or (b) the input is incomplete (valid so far but incomplete without more bytes). InvalidInputError was supposed to represent (a). This seems like the same thing, yes?

---

My idea was to classify the errors into a third case where the input does not even begin with valid RESP marker (`+`, `-`, `:`, `$`, `*`). 
But yeah, this does seem redundant. I'll go over and change this.